sl_init_mod.f90

```diff
@@ -200,12 +204,15 @@ end subroutine write_nf90
    subroutine read_nf90(data_slm, filename, filepath, suffix, fext)
 
       character (len = *), intent(in) :: filename, filepath !file name and path
-      real, dimension(2*nglv,nglv) :: data_temp !temp. variable name in the SLM in which nc data will be stored
-      real, dimension(nglv,2*nglv), intent(out) :: data_slm
+      real, dimension(:,:), allocatable :: data_temp !temp. variable name in the SLM in which nc data will be stored
+      real, dimension(:,:), intent(out) :: data_slm
       character (len = *), optional ::  suffix
       character (len = *), optional :: fext
       integer :: ncid, varid
 
+      allocate (data_temp(2*nglv,nglv))
```

Comment:
You need to add a corresponding `deallocate` at the end of the routine to avoid a memory leak